react/player/src/asset/index.tsx

```diff
@@ -49,11 +49,30 @@ export const ReactAsset = (
     throw Error(`Asset is missing type for ${info}`);
   }
 
+  if (!registry || registry.isRegistryEmpty()) {
+    throw Error(`No asset found in registry. This could happen for one of the following reasons: \n
+      1. You might have no assets registered or no plugins added to the Player instance. \n
+      2. You might have mismatching versions of React Asset Registry Context. \n
+      See https://player-ui.github.io/latest/tools/cli#player-dependency-versions-check for tips about how to debug and fix this problem`);
+  }
+
   const Impl = registry?.get(unwrapped);
 
+  const matchList: object[] = [];
+
+  registry.forEach((asset) => {
+    matchList.push(asset.key);
+  });
```

Comment:
Can we compute this only in the error case so we don't do this work if there is no error?